sessions/react-state-2/table-reservation/README.md

```diff
@@ -0,0 +1,45 @@
+# React State 2: Table Reservation
+
+In this challenge, there is a simple app to reserve a table at a restaurant.
+
+Have a look at the [`Counter Component`](./src/components/Counter.js): there is a state which gets updated when the buttons are clicked.
+
+In the [`src/App.js`](./src/App.js) file, however, there is a sentence saying "You are going to reserve a table for 2 people." The number should not be hard coded, but equal the current people count and need to be updated on every button click.
+
+## Task
+
+Lift the `people` state up so that
+
+1. it still gets updated by the `Counter` component and
+2. the actual people count is dynamically displayed in the `App` component.
+
+You can use the following hints as guideline:
+
+- Move the `people` state to the `App` component.
+- In the `App` component, you want to have two helper functions to `handle` adding and subtracting a person from the `people` state.
+- Pass your `handle` functions to the `Counter` component:
+  - use one prop for each of them;
+  - the prop should have the same name as the `handle` function, but should start with `on` instead of `handle` (Hint: `onDelete={handleDelete}`).
+- In the `Counter` component
+  - receive the two newly created props and
+  - pass them to the correct button's `onClick` prop. Make sure **not** to use an anonymous function here.
```

Comment:
why?

Reply:
Finde es fein, das mit inline functions zu lösen:
```js
      <Counter
        onIncrement={() => setPeople(people + 1)}
        onDecrement={() => setPeople(people - 1)}
      />
```

Reply:
I see the point. I think it's then better to delete the second hint about the anonymous function, like so:
```suggestion
  - pass them to the correct button's `onClick` prop.
```